addons/event_sms/models/event_registration.py

```diff
@@ -1,13 +1,63 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import api, models
+from odoo.addons.phone_validation.tools import phone_validation
 
 
 class Registration(models.Model):
     _inherit = 'event.registration'
 
+    @api.model_create_multi
+    def create(self, values_list):
+        """ Override to try to format mobile or phone numbers, as it is used
+        notably to send reminders. Having international numbers eases process
+        of preparing and sending SMS. """
+        registrations = super().create(values_list)
+        for registration in registrations.filtered(
+            lambda reg: reg.mobile or reg.phone
+        ):
+            for fname in {'mobile', 'phone'}:
+                new_number = registration._phone_format(fname)
+                if new_number and new_number != registration[fname]:
+                    registration[fname] = new_number
```

Comment:
It seems like the ORM already doesn't write if the value hasn't changed and is cached anyway so maybe not necessary. But works either way :man_shrugging:

Reply:
Mmmh ... due to some history with the ORM I tend to be careful when updating records :D . Would keep it unless you really hate that if.

Reply:
Something something... Better safe than query
Let's keep the check :)